src/renderer/prefComponents/spellchecker/index.vue

```diff
@@ -2,32 +2,32 @@
   <div class="pref-spellchecker">
     <h4>Spelling</h4>
     <bool
-      description="Whether the (experimental) spell checker is enabled."
+      description="Enable spell checker (experimental)."
       :bool="spellcheckerEnabled"
       :onChange="handleSpellcheckerEnabled"
     ></bool>
     <separator></separator>
     <bool
-      description="Whether Hunspell is used instead of the OS spell checker on macOS and Windows 10. The change takes effect after restarting Mark Text and for new editor windows."
+      description="Use Hunspell instead of system spell checker on macOS and Windows 10. (Restart required.)
       :bool="spellcheckerIsHunspell"
       :disable="!isOsSpellcheckerSupported || !spellcheckerEnabled"
       :onChange="value => onSelectChange('spellcheckerIsHunspell', value)"
     ></bool>
     <bool
-      description="Don't underline spelling mistakes. You can still correct spelling mistakes via the right click menu."
+      description="Hide marks on spelling errors."
       :bool="spellcheckerNoUnderline"
       :disable="!spellcheckerEnabled"
       :onChange="value => onSelectChange('spellcheckerNoUnderline', value)"
     ></bool>
     <bool
-      description="Automatically identify the used language. This feature is currently unavailable when using Hunspell or if mistake underlining is disabled."
+      description="Automatically detect document language. (Unavailable if Hunspell is used or if spelling errors are not marked.)
```

Comment:
```suggestion
      description="Automatically detect document language (spelling errors must be marked and macOS only).
```

Reply:
We certainly can revise this line, but grammatically the suggested version means "spelling errors must be marked and *spelling errors must be* macOS only".

Reply:
I'll submit a PR to disable the checkbox if the requirements aren't met, so we can remove the second part completely.

Reply:
The user generally would not know why the box is disabled. Marking spelling errors and detecting language is not an obvious relationship. (In fact it seems to be a strange limitation. I would understand that a dictionary needs to be installed, but not why the UI feature of marking is required.)

Reply:
> In fact it seems to be a strange limitation. I would understand that a dictionary needs to be installed, but not why the UI feature of marking is required.

You can enable the feature but it isn't applied to the spell checker. First of all, language detection is currently only available on macOS because of a bug in Electron that we cannot spawn another (worker) thread and macOS provides this feature integrated in the OS spell checker. Of course, we could create a C++ wrapper library where the language detection library (that's written in C++ too) runs on another thread but I don't have time to do so. Why another thread? Because language detection comsumes a lot of CPU and we don't want to block the renderer process.
Marking spelling errors is only required by macOS spell checker because the spell checker analyses the typed words and change the language accordantly. If no spelling errors are marked, no input can be analysed by macOS and the spell checker is unable to detect the language. We could workaround this issue by enabling the language detection that is used on Linux and Windows but that is disabled due to a bug in Electron. Basically, this "feature" cause more problems than it would solve :smile: .

Reply:
I understand your explanation concerning technical limitations. The reason for my comment was that nothing in your  explanation is obvious to the end user (or even to me before learning it from you). If the user would want that language is detected *and could get the option by enabling the spelling feature*, then knowing that relationshp is helpful.

Reply:
Hearing about the headache you are facing I feel inclined to suggest that you just move language detection into a feature branch and remove it from the release until the application and the other dependencies become more mature. But that is only a suggestion outside the scope of this PR.